src/components/AnonymousChat/child/ChatItemTargetText.tsx

```diff
@@ -71,27 +79,42 @@ const ChatItemTargetText = ({
   time = '4h',
   isContinuous = false,
   message = '',
+  attachments = [],
   avatar
 }: ChatItemMyTextProps) => {
   const renderAvatar = React.useCallback(() => {
     if (isContinuous) return <View style={styles.avatar} />;
     return <View style={styles.mr8}>{avatar}</View>;
   }, []);
 
+  const handleTextContainerStyle = () => {
+    const isAttachment = attachments.length > 0;
+    const paddingStyle = {
+      paddingLeft: isAttachment ? BUBBLE_LEFT_PADDING_ATTACHMENT : BUBBLE_LEFT_PADDING,
+      paddingRight: isAttachment ? BUBBLE_RIGHT_PADDING_ATTACHMENT : BUBBLE_RIGHT_PADDING
+    };
+    return [styles.textContainer, paddingStyle];
+  };
+
   return (
     <View style={styles.chatContainer}>
       {renderAvatar()}
-      <View style={styles.textContainer}>
+      <View style={handleTextContainerStyle()}>
         {!isContinuous && (
-          <View testID="chat-item-user-info" style={styles.chatTitleContainer}>
+          <View
+            testID="chat-item-user-info"
+            style={[styles.chatTitleContainer, attachments.length > 0 ? {marginBottom: 4} : {}]}>
             <Text style={styles.userText}>{username}</Text>
             <View style={styles.dot} />
             <Text style={styles.timeText}>{time}</Text>
           </View>
         )}
-        <Text testID="chat-item-message" style={styles.text}>
-          {message}
-        </Text>
+        {attachments.length > 0 && <ChatItemAttachment attachments={attachments} />}
+        {attachments.length <= 0 && (
+          <Text testID="chat-item-message" style={styles.text}>
+            {message}
+          </Text>
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code has been refactored to handle attachments in chat messages. This is a significant change and should be thoroughly tested to ensure it works as expected. The logic for rendering the avatar, handling text container style, and rendering the message or attachment seems correct.

One thing to note is that the `renderAvatar` function and `handleTextContainerStyle` function are redefined on every render. Consider using `React.useMemo` instead of `React.useCallback` for these functions to avoid unnecessary re-renders.

Also, make sure to handle potential errors when dealing with attachments. For example, what happens if an attachment fails to load? It might be beneficial to have some error handling or fallback in place.

```diff
-   avatar
+   attachments = [],
+   avatar
 }: ChatItemMyTextProps) => {
-   const renderAvatar = React.useCallback(() => {
+   const renderAvatar = React.useMemo(() => {
     if (isContinuous) return <View style={styles.avatar} />;
     return <View style={styles.mr8}>{avatar}</View>;
-   }, []);
+   }, [isContinuous, avatar]);

-   const handleTextContainerStyle = () => {
+   const handleTextContainerStyle = React.useMemo(() => {
     const isAttachment = attachments.length > 0;
     const paddingStyle = {
       paddingLeft: isAttachment ? BUBBLE_LEFT_PADDING_ATTACHMENT : BUBBLE_LEFT_PADDING,
       paddingRight: isAttachment ? BUBBLE_RIGHT_PADDING_ATTACHMENT : BUBBLE_RIGHT_PADDING
     };
     return [styles.textContainer, paddingStyle];
-   };
+   }, [attachments]);

   return (
     <View style={styles.chatContainer}>
       {renderAvatar()}
      <View style={handleTextContainerStyle()}>
         {!isContinuous && (
          <View
            testID="chat-item-user-info"
            style={[styles.chatTitleContainer, attachments.length > 0 ? {marginBottom: 4} : {}]}>
             <Text style={styles.userText}>{username}</Text>
             <View style={styles.dot} />
             <Text style={styles.timeText}>{time}</Text>
           </View>
         )}
        {attachments.length > 0 && <ChatItemAttachment attachments={attachments} />}
        {attachments.length <= 0 && (
          <Text testID="chat-item-message" style={styles.text}>
            {message}
          </Text>
        )}
       </View>
     </View>
   );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->